Notifications API actions/maxActions and friends (#44184)
mdn/content: files/en-us/web/api/notification/maxactions_static/index.md

browser-compat: api.Notification.maxActions_static
---

{{APIRef("Web Notifications")}}{{SecureContext_Header}}{{SeeCompatTable}} {{AvailableInWorkers}}

> The **`maxActions`** read-only static property of the {{domxref("Notification")}} interface returns the maximum number of actions supported by the device and the user agent.

Reviewer: ```suggestion
The **`maxActions`** read-only static property of the {{domxref("Notification")}} interface returns the maximum number of actions the device and user agent will display per notification.
```

This is more what it sounds like it is, from reading the `actions` page.

Author: Yes, support means "that can be displayed". The limit is really the device, since that is what provides the notification. I think I'll do a small simplification to remove "what" creates the limit.

```suggestion
The **`maxActions`** read-only static property of the {{domxref("Notification")}} interface returns the maximum number of actions that can be displayed in a notification.
```

Reviewer: OK, makes sense.

## Value

> An integer number.

Reviewer: ```suggestion
An integer.
```

## Description

Notification actions are buttons or controls that appear within [persistent notifications](/en-US/docs/Web/API/Notifications_API#persistent_and_non-persistent_notifications).
Actions are set using the [`actions`](/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification#actions) option of the second argument of the {{domxref("ServiceWorkerRegistration.showNotification", "showNotification()")}} method.

Browsers typically limit the maximum number of actions they will display for a particular notification.

> This property returns that limit, which is the maximum number of elements in the {{domxref("Notification.actions")}} array that will be respected by the user agent.

Reviewer: ```suggestion
The `maxActions` property returns that limit, which is the maximum number of elements in the {{domxref("Notification.actions")}} array that will be respected by the user agent.
```
